shuup/core/models/_service_behavior.py

```diff
@@ -163,7 +164,7 @@ class GroupAvailabilityBehaviorComponent(ServiceBehaviorComponent):
     groups = models.ManyToManyField("ContactGroup", verbose_name=_("groups"))
 
     def get_unavailability_reasons(self, service, source):
-        if not source.customer.pk:
+        if not issubclass(type(source.customer), AnonymousContact) and not source.customer.pk:
```

Comment:
If this is indeed what we want (still waiting for Panu's answer, since IMHO we should just remove the support for unsaved non-anonymous contacts in `request.customer`), then use `if source.customer and not source.customer.pk`. It will work, since `AnonymousContact` will evaluate as false in the `if`.